File: src/test/java/de/kevcodez/metronom/test/model/station/StationProviderTest.java
package de.kevcodez.metronom.test.model.station;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.greaterThan;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.not;
import static org.hamcrest.Matchers.notNullValue;

import de.kevcodez.metronom.model.station.Station;
import de.kevcodez.metronom.model.station.StationProvider;

import java.util.List;

import org.junit.Before;
import org.junit.Test;

public class StationProviderTest {

  private StationProvider stationProvider;

  @Before
  public void before() {
    stationProvider = new StationProvider();
    stationProvider.constructStations();
  }

  @Test
  public void shouldProvideStations() {
    List<Station> stations = stationProvider.getStations();
    assertThat(stations.size(), greaterThan(0));
  }

  @Test
  public void shouldFindStationByCode() {
    Station station = stationProvider.findStationByCode("AMA");

    assertThat(station, notNullValue());
    assertThat(station.getName(), is("Maschen"));
    assertThat(station.getCode(), is("AMA"));
  }

  @Test
  public void shouldFindStationByName() {
    Station station = stationProvider.findStationByName("Harburg");

    assertThat(station, notNullValue());
    assertThat(station.getName(), is("Harburg"));
  }

  @Test
  public void shouldFindStationByAlternativeName() {
    Station station = stationProvider.findStationByName("Hamburg-Harburg");

    assertThat(station, notNullValue());
    assertThat(station.getName(), is("Harburg"));
  }

  @Test
  public void shouldHaveEqualStations() {
    Station stationByCode = stationProvider.findStationByCode("AMA");
    Station stationByName = stationProvider.findStationByName("Maschen");

    assertThat(stationByCode, is(stationByName));
    assertThat(stationByCode.hashCode(), is(stationByName.hashCode()));

    Station differentStation = stationProvider.findStationByName("Harburg");
    assertThat(stationByCode, not(differentStation));
    assertThat(stationByCode.hashCode(), not(differentStation.hashCode()));
  }

}

File: src/test/kotlin/de/kevcodez/metronom/test/model/station/StationProviderTest.kt
package de.kevcodez.metronom.test.model.station

import de.kevcodez.metronom.model.station.Station
import de.kevcodez.metronom.provider.StationProvider
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.*
import org.junit.jupiter.api.Test

class StationProviderTest {

    @Test
    fun shouldProvideStations() {
        val stations = StationProvider.getStations()
        assertThat(stations.size, greaterThan(0))
    }

    @Test
    fun shouldFindStationByCode() {
        val station = StationProvider.findStationByCode("AMA")

        assertThat(station, notNullValue())
        assertThat<String>(station.name, `is`("Maschen"))
        assertThat<String>(station.code, `is`("AMA"))
    }

    @Test
    fun shouldFindStationByName() {
        val station = StationProvider.findStationByName("Harburg")

        assertThat<Station>(station, notNullValue())
        assertThat<String>(station!!.name, `is`("Harburg"))
    }

    @Test
    fun shouldFindStationByAlternativeName() {
        val station = StationProvider.findStationByName("Hamburg-Harburg")

        assertThat<Station>(station, notNullValue())
        assertThat<String>(station!!.name, `is`("Harburg"))
    }

    @Test
    fun shouldHaveEqualStations() {
        val stationByCode = StationProvider.findStationByCode("AMA")
        val stationByName = StationProvider.findStationByName("Maschen")

        assertThat(stationByCode, `is`(stationByName))
        assertThat(stationByCode.hashCode(), `is`(stationByName!!.hashCode()))

        val differentStation = StationProvider.findStationByName("Harburg")
        assertThat(stationByCode, not(differentStation))
        assertThat(stationByCode.hashCode(), not(differentStation!!.hashCode()))
    }

}
